trio/_core/tests/test_run.py

```diff
@@ -1696,74 +1696,6 @@ async def test_current_effective_deadline(mock_clock):
     assert _core.current_effective_deadline() == inf
 
 
-# @coroutine is deprecated since python 3.8, which is fine with us.
-@pytest.mark.filterwarnings("ignore:.*@coroutine.*:DeprecationWarning")
-def test_nice_error_on_bad_calls_to_run_or_spawn():
-    def bad_call_run(*args):
```

Comment:
You've removed the test that passing these things to run() or start_soon() actually results in an error (ie, that your now factored-out helper actually gets called). You don't need to duplicate the full suite of tests, but you should still pass one or two error-provoking arguments and ensure that errors get produced.

Reply:
Ok, added test for passing a sync function and an async generator.